frontend/src/Components/WorkstationView.tsx

```diff
@@ -57,42 +59,60 @@ const GRAPH_FIELDS = enumVals(GraphField);
 
 export const WorkstationView = ({ hostname }: { hostname: string }) => {
   const [, setPs] = useSearchParams();
+  const { allStats } = useStats();
 
-  return (
-    <Modal
-      size="xl"
-      isOpen={hostname !== null}
-      onClose={() => {
-        setPs((ps) => {
-          ps.delete("selected");
-          return ps;
-        });
-      }}
-      scrollBehavior="inside"
-    >
-      <ModalOverlay bg="blackAlpha.300" backdropFilter="blur(15px)" />
-      <ModalContent minWidth="80%" minHeight="80%">
-        <ModalHeader>{`${hostname}`}</ModalHeader>
-        <ModalCloseButton />
-        <ModalBody>
-          <SimpleGrid columns={2} spacing={5}>
-            <StatsTable hostname={hostname}></StatsTable>
-            <StatsGraph hostname={hostname} />
-          </SimpleGrid>
-        </ModalBody>
-      </ModalContent>
-    </Modal>
-  );
+  return validationElim(allStats, {
+    success: (stats) => {
+      const { wstat, name } = stats.flatMap((g) =>
```

Comment:
If someone deletes the workstation we are on the page for, I think the `[0]` will error. Do we want to handle this case (ig deselect the workstation)?

Reply:
Opened an issue https://github.com/gpuctl/gpuctl/issues/251